benchmarks/benchmarks/sparse_csgraph_disjoint_set.py

```diff
@@ -19,18 +19,41 @@ def setup(self, n):
         self.nodes = np.unique(self.edges)
         self.disjoint_set = DisjointSet(self.nodes)
 
-        self.premerged = DisjointSet(self.nodes)
+        self.pre_merged = DisjointSet(self.nodes)
         for a, b in self.edges:
-            self.premerged.merge(a, b)
+            self.pre_merged.merge(a, b)
+
+        self.pre_merged_found = DisjointSet(self.nodes)
+        for a, b in self.edges:
+            self.pre_merged_found.merge(a, b)
+        for x in self.nodes:
+            self.pre_merged_found[x]
 
     def time_merge(self, n):
         dis = self.disjoint_set
         for a, b in self.edges:
             dis.merge(a, b)
 
+    def time_merge_already_merged(self, n):
+        dis = self.pre_merged
+        for a, b in self.edges:
+            dis.merge(a, b)
+
     def time_find(self, n):
-        dis = self.premerged
+        dis = self.pre_merged
+        return [dis[i] for i in self.nodes]
+
+    def time_find_already_found(self, n):
+        dis = self.pre_merged_found
         return [dis[i] for i in self.nodes]
 
     def time_contains(self, n):
-        return self.nodes[0] in self.premerged
+        # Test for presence
+        assert self.nodes[0] in self.pre_merged
+        assert self.nodes[n // 2] in self.pre_merged
+        assert self.nodes[-1] in self.pre_merged
+
+        # Test for absense
+        assert None not in self.pre_merged
+        assert "dummy" not in self.pre_merged
+        assert (1, 2, 3) not in self.pre_merged
```

Comment:
can these be moved to a separate method? I want the benchmark to verify that they take the same amount of time.

Reply:
Got it. I have split the test in two.

On my laptop they do not take the same amount of time:
```
[ 58.33%] ··· sparse_csgraph_disjoint_set.Bench.time_absence                                                                ok
[ 58.33%] ··· ======= ==========
                 n              
              ------- ----------
                100    699±20ns 
                1000   678±30ns 
               10000   678±20ns 
              ======= ==========

[ 66.67%] ··· sparse_csgraph_disjoint_set.Bench.time_contains                                                               ok
[ 66.67%] ··· ======= =============
                 n                 
              ------- -------------
                100    1.25±0.02μs 
                1000   1.27±0.01μs 
               10000   1.36±0.05μs 
              ======= =============
```